### ISSUE TITLE
question about label

### ISSUE THREAD

**Issue by Huuush** [Fan9/SeesawLoss_pytorch#1]
大佬好，小白求教。想请问一下你实现中的label是针对每个propose的Anchor的么？ mmdet里，loss_cls部分的输入应该是所有Anchor的logits和targets，shape为[N, C]（N为anchor数量，C为class数量）。在你的实现里`targets = F.one_hot(targets, self.num_labels)`，我是不是应该循环所有的anchor，把单个anchor的target传到seesaw里面？ 先感谢一下！

**Comment by Fan9** [comment]
不用，你直接把`targets = F.one_hot(targets, self.num_labels)`这一句注释掉，然后把logits和labels传进来就好了。
当时为了方便和其他loss对比，在实现时forward的参数和pytorch官方的提供的loss保持一致，labels是shape是[N]。而seesaw loss实现时需要用到one-hot之后的labels。pytorch的forward参数如下：
![捕获1](https://user-images.githubusercontent.com/35786245/112425985-0f744b80-8d72-11eb-898a-bd1ac801cb15.PNG)

**Comment by Huuush** [comment]
噢！ 我懂了！感谢大佬！

**Comment by Huuush** [comment]
大佬 我又来打扰了。请问你有尝试在mmdet里测过这个实现吗？ 我尝试把它加在现有的loss后面，结果梯度炸了...
